proxy_docker/app/script/call_lightningd.sh

```diff
@@ -58,9 +72,13 @@ ln_create_invoice() {
     if [ -n "${connectstring}" ]; then
       data="${data}\"connectstring\":\"${connectstring}\","
     fi
-    data="${data}\"callback_url\":\"${callback_url}\","
+    if [ "${callback_url}" != "null" ]; then
```

Comment:
IMO  `callback_url` and `msatoshi` fields should always be returned in the confirmation payload (even if values are null). Having the schema of the payload change creates headaches downstream.

Reply:
Do you mean, for example, if the caller doesn't supply callback_url, we should anyway include it in the response with a `null` value?

Reply:
Or should we, everywhere, differentiate `null` value and absence of the field?

Reply:
In my personal and humble opinion, Yes. i would aim for consistency, so would return ```callback_url=null``` if not provided rather than not include the field in the payload :)

Reply:
Even when `callback_url` is not required?

Reply:
Optional fields, that are empty should be left out to avoid visual clutter, IMHO null values and empty strings should remove the key completely. When in doubt, I ALWAYS choose readability. If you consistently leave out empty values it still is consistent. ;-)

Reply:
Interesting.  I didn't know that it was a thing to think to include optional stuff without value in a response.  I'm surprised to see discussions about that!  For example:

https://softwareengineering.stackexchange.com/questions/343710/use-empty-string-null-or-remove-empty-property-in-api-request-response

Maybe I'm too old and because I programmed on less-than-half-a-MB-of-RAM computers, but it doesn't really make sense to me to include information saying "this optional value has no value" when it's actually set as optional in the schema.  It's not really optimal to include a 1-MB JSON response with a zillion optional fields with null values.  I'd rather send an empty JSON that will still be valid wrt to the schema and tell everything needed to be told about the response.

Reply:
What you're saying makes 100% and I would add that it is contextual as you had mentioned. 
In this context since the behavior of the call is affected by the optional parameters provided (or lack there of) by the user `msatoshi` and `callback_url` , including the return of null in the payload *explicitly* informs the user of the outcome of his call. For me that is clarity and consistency.